src/PSDMatrices.jl

```diff
@@ -42,6 +42,15 @@ function logdet(M::PSDMatrix)
     return 2 * logdet(M.R)
 end
 
+function diag(M::PSDMatrix)
+    out = similar(M.R, size(M.R, 2))
+    @simd ivdep for i in axes(M.R, 2)
```

Comment:
`@simd` declares the loop as re-orderable by the compiler. To be honest, I don't fully understand the `ivdep` part, but I found it throughout OrdinaryDiffEq.jl and I have been recommended to use it :sweat_smil